.github/workflows/ci.yml

```diff
@@ -54,15 +62,40 @@ jobs:
           coverage: codecov
 
         - name: Python 3.12 (Windows)
-          windows: py312-test
+          windows: py312-test-xdist
           posargs: -v
 
         - name: Python 3.12 (OSX)
-          macos: py312-test
+          macos: py312-test-xdist
           posargs: -v
 
         - name: Python 3.12 with coverage
           linux: py312-test-cov
           posargs: -v
           coverage: codecov
 
+        - name: Python 3.11 (Windows with dev dependencies)
+          windows: py311-test-xdist-devdeps
+          posargs: -v
+
+        - name: Python 3.11 (OSX with dev dependencies)
+          macos: py311-test-xdist-devdeps
+          posargs: -v
+          pytest-results-summary: true
+
+        - name: Python 3.11 (Linux with dev dependencies)
+          linux: py311-test-xdist-devdeps
+          posargs: -v
+
+        - name: Python 3.12 (Windows with dev dependencies)
+          windows: py312-test-xdist-devdeps
+          posargs: -v
+
+        - name: Python 3.12 (OSX with dev dependencies)
+          macos: py312-test-xdist-devdeps
+          posargs: -v
+
+        - name: Python 3.12 (Linux with dev dependencies)
+          linux: py312-test-xdist-devdeps
+          posargs: -v
+          pytest-results-summary: true
```

Comment:
I recommend that you run the `devdeps` tests only with `py312`.  The other versions are unnecessary.

Reply:
In fact, only Linux makes sense for devdeps. Not all packages build dev wheels for all the architectures.